guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryPresenter.java

```diff
@@ -0,0 +1,211 @@
+
+    @Inject
+    private PlaceManager placeManager;
+
+    private Map<String, OrganizationalUnit> availableOrganizationalUnits = new HashMap<String, OrganizationalUnit>();
+
+    @PostConstruct
+    public void init() {
+        isOuMandatory = isOuMandatory();
+        view.init(this, isOuMandatory);
+        populateOrganizationalUnits();
+    }
+
+    @Override
+    public void handleCancelClick() {
+        view.hideForm();
+    }
+
+    @Override
+    public void handleCloneClick() {
+        if ( view.isGitUrlEmpty() ) {
+            view.setUrlGroupType( ControlGroupType.ERROR );
+            view.setUrlHelpMandatory();
+            return;
+        } else if ( !URIUtil.isValid(view.getGitUrl()) ) {
+            view.setUrlGroupType( ControlGroupType.ERROR );
+            view.setUrlHelpInvalidFormat();
+            return;
+        } else {
+            view.setUrlGroupType( ControlGroupType.NONE );
+        }
+        final String organizationalUnit = view.getOrganizationalUnit(view.getSelectedOrganizationalUnit());
+        if ( isOuMandatory && !availableOrganizationalUnits.containsKey( organizationalUnit ) ) {
+            view.setOrganizationalUnitGroupType( ControlGroupType.ERROR );
+            view.setOrganizationalUnitHelpMandatory();
+            return;
+        } else {
+            view.setOrganizationalUnitGroupType( ControlGroupType.NONE );
+        }
+
+        if ( view.isNameEmpty() ) {
+            view.setNameGroupType( ControlGroupType.ERROR );
+            view.setNameHelpMandatory();
+            return;
+        } else {
+            repositoryService.call( new RemoteCallback<String>() {
+                @Override
+                public void callback( String normalizedName ) {
+                    if ( !view.getName().equals( normalizedName ) ) {
+                        if ( !view.showAgreeNormalizeNameWindow(normalizedName) ) {
+                            return;
+                        }
+                        view.setName(normalizedName);
+                    }
+
+                    lockScreen();
+
+                    final String scheme = "git";
+                    final String alias = view.getName().trim();
+                    final String origin = view.getGitUrl();
+                    final String username = view.getUsername().trim();
+                    final String password = view.getPassword().trim();
+                    final Map<String, Object> env = new HashMap<String, Object>( 3 );
+                    env.put( "username", username );
+                    env.put( "crypt:password", password );
+                    env.put( "origin", origin );
+
+                    repositoryService.call( new RemoteCallback<Repository>() {
+                                                @Override
+                                                public void callback( Repository o ) {
+                                                    view.closeBusyPopup();
```

Comment:
We have an interesting scenario here. If you show the form once and successfully create a Repository the form remains "locked" due to the call to ```lockScreen()``` above. The screen is successfully "unlocked" in the ```ErrorCallback``` but not in the ```RemoteCallback```. You do not "unlock" the screen in the ```showForm()``` method so I think there's a risk the popup could remain locked.